config-model/src/test/java/com/yahoo/searchdefinition/SummaryTestCase.java

```diff
@@ -0,0 +1,47 @@
+package com.yahoo.searchdefinition;
+
+import com.yahoo.config.application.api.DeployLogger;
+import com.yahoo.searchdefinition.parser.ParseException;
+import com.yahoo.vespa.model.test.utils.DeployLoggerStub;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.logging.Level;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests summary validation
+ *
+ * @author bratseth
+ */
+public class SummaryTestCase {
+
+    @Test
+    public void testMemorySummary() throws IOException, ParseException {
+        DeployLoggerStub logger = new DeployLoggerStub();
+        SearchBuilder.createFromFile("src/test/examples/memorysummary.sd", logger);
```

Comment:
I prefer that small sd files like these are tested with SearchBuilder.createFromString("...") and the sd file inline in the test. Makes it a lot easier to understand the test. Goes for all 3 tests.